### PR TITLE
NOT READY inspected code

### PR DESCRIPTION
**Pull request by aweng98**
Cleaned up Scala code warnings found by IntelliJ: Analyze > Inspect Code

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aweng98** [comment]
After cleanup, Analyzed results are:
* . Properties Files - 1 warning
* . Scala General - 1 weak warning
* . Scaladoc - 4 warnings

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataLibrarySpec.scala [hunk 1]
```diff
@@ -136,9 +136,8 @@ class DataLibrarySpec extends FreeSpec with WebBrowserSpec with UserFixtures wit
                 val page = new DataLibraryPage().open
 
                 // entering multiple tags in Tags input-field
-                val expectedTags = attrTag.get("tag:tags").get
+                val expectedTags = attrTag("tag:tags")
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataLibrarySpec.scala]: this change may not work. ?? wait for test results to confirm

File: automation/project/Dependencies.scala [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import sbt._
 
 object Dependencies {
-  val jacksonV = "2.9.5"
+  val jacksonV = "2.9.0"
```
Comment (helgridly) [inline automation/project/Dependencies.scala:4]: this is a downgrade: why?

File: automation/project/Dependencies.scala [hunk 3]
```diff
@@ -43,7 +43,7 @@ object Dependencies {
     "com.typesafe.akka"   %%  "akka-http"           % akkaHttpV,
     "com.typesafe.akka"   %%  "akka-testkit"        % akkaV     % "test",
     "com.typesafe.akka"   %%  "akka-slf4j"          % akkaV,
-    "org.specs2"          %%  "specs2-core"   % "3.7"  % "test",
+    //"org.specs2"          %%  "specs2-core"   % "3.7"  % "test",
```
Comment (helgridly) [inline automation/project/Dependencies.scala:46]: why?

File: automation/project/Settings.scala [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import Dependencies._
 import sbt.Keys._
-import sbt._
+import sbt.{Def, _}
```
Comment (helgridly) [inline automation/project/Settings.scala:3]: importing `Def` as well shouldn't be necessary, `_` means "everything in the package"

File: automation/project/Settings.scala [hunk 5]
```diff
@@ -38,18 +38,18 @@ object Settings {
   )
 
   //common settings for all sbt subprojects
-  val commonSettings =
+  val commonSettings: Seq[Def.Setting[_]] =
     commonBuildSettings ++ testSettings ++ List(
     organization  := "org.broadinstitute.dsde.firecloud",
-    scalaVersion  := "2.11.8",
+    scalaVersion  := "2.12.6",
```
Comment (helgridly) [inline automation/project/Settings.scala:44]: this is a big change: are you also intending to upgrade to 2.12 here?

File: automation/project/Settings.scala [hunk 6]
```diff
@@ -38,18 +38,18 @@ object Settings {
   )
 
   //common settings for all sbt subprojects
-  val commonSettings =
+  val commonSettings: Seq[Def.Setting[_]] =
```
Comment (helgridly) [inline automation/project/Settings.scala:41]: IMO this particular inspection is busy-work, so i wouldn't stress too much about adding type annotations for everything

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/LibraryData.scala [hunk 7]
```diff
@@ -37,7 +37,7 @@ object LibraryData {
      "library:technology"->Seq("is an optional","array attribute"),
      "_discoverableByGroups"->Seq("all_broad_users"))
 
-   val metadataORSP = metadataBasic ++ Map(
+   val metadataORSP: Map[String, Any] = metadataBasic ++ Map(
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/fixture/LibraryData.scala:40]: why this one but not `metadataBasic` too? (though i don't think you need either)

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala [hunk 8]
```diff
@@ -97,7 +97,7 @@ class DataLibraryPage(implicit webDriver: WebDriver) extends BaseFireCloudPage
     libraryTable.getRows
   }
 
-  case class RequestAccessModal(override implicit val webDriver: WebDriver) extends MessageModal {
+  case class RequestAccessModal() extends MessageModal {
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/library/DataLibraryPage.scala:100]: what was the inspection for this change? was it that the val is already defined in the superclass?

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/RegistrationPage.scala [hunk 9]
```diff
@@ -50,7 +50,4 @@ class RegistrationPage(implicit webDriver: WebDriver) extends AuthenticatedPage
     registerButton.doClick()
   }
 
-  private def registrationCompleteMessageIsPresent(): Boolean = {
-    find(registrationCompleteMessage).isDefined
-  }
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/RegistrationPage.scala:55]: dead code?

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala [hunk 10]
```diff
@@ -73,7 +73,6 @@ class GoogleSignInPopup(implicit webDriver: WebDriver) extends WebBrowser with W
     * Signs in to Google to authenticate for FireCloud.
     */
   def signIn(email: String, password: String): Unit = {
-    val chooseAccount = find(id("identifierLink")).filter(_.isDisplayed) foreach { click on _ }
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala:76]: i'm assuming this turned out to be unnecessary, but it's definitely doing things

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/notebooks/WorkspaceNotebooksPage.scala [hunk 11]
```diff
@@ -129,10 +129,10 @@ class CreateClusterModal(implicit webDriver: WebDriver) extends OKCancelModal("c
     clickOk()
   }
 
-  def addLabels(labels: Map[String, String]) = {
+  def addLabels(labels: Map[String, String]): Unit = {
     optionalSettingsArea.ensureExpanded()
     var labelIndex = 0
-    labels.map{ label =>
+    labels.foreach{ label =>
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/notebooks/WorkspaceNotebooksPage.scala:135]: this is a good one to watch out for :+1:

**Review by helgridly — COMMENTED** [review]
What's with the jackson downgrade and scala upgrade?

**Pull request closed without merging**